### PR TITLE
Fixing wrong reference

### PR DESCRIPTION
**Pull request by xtrusia**
Fixing wrong reference

relation_id is in Endpoint class, but self.relation. returns relation.

so it should be self.relation_id instead of self.relation.relation_id

Actually I think this haven't used before, as LP[1] is the cause of this implementation. ( prior patch was : https://github.com/juju-solutions/charms.reactive/pull/232 )

Im trying to make patch for app relation data. but prior patch has bug. 
This bug is not public I think but when I tries to use prior patch, I face issue.

[1]
LP : https://bugs.launchpad.net/charm-kubernetes-master/+bug/1899706

### CODE REVIEW

**Comment by ajkavanagh** [comment]
> Fixing wrong reference

Please could you supply a bit more context to the PR?  e.g. what bug is it fixing?  There should also be a unit test to verify the change (particularly if there was no coverage of the code previously).

**Comment by xtrusia** [comment]
> > Fixing wrong reference
> 
> Please could you supply a bit more context to the PR? e.g. what bug is it fixing? There should also be a unit test to verify the change (particularly if there was no coverage of the code previously).

**Comment by xtrusia** [comment]
accidently i clicked closed button.. sorry

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python: [3.5, 3.6, 3.7, 3.8, 3.9]
+        python: [3.6, 3.7, 3.8, 3.9]
```
Comment (ajkavanagh) [inline .github/workflows/tests.yaml:9]: Technically, this would be a different PR; essentially, you're removing support for py35 from the library which would be mixed in with this commit.

**Review by ajkavanagh — COMMENTED** [review]
Still would like to see a bug for this issue, please? e.g. Is this code silently breaking other charms that are using the same function `to_publish_raw`?  What charm is it failing on?  Thanks.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (xtrusia) [reply .github/workflows/tests.yaml:9]: I removed this because github action ( testing ) failed.

Comment (xtrusia) [reply .github/workflows/tests.yaml:9]: thanks and i updated top comment

**Review by ajkavanagh — CHANGES_REQUESTED** [review]
Please could you remove the *.swo files from the repository.  In general, if your editor generates these, you can place `*.swo` in your global `.gitignore` and they will never be included in commits.  Thanks.

**Comment by xtrusia** [comment]
thanks, I removed temp file and re-added py 3.5 for testing.

testing for 3.5 will fail but didn't remove it as you said it is needed for supporting xenial

Thanks.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (ajkavanagh) [reply .github/workflows/tests.yaml]: > I removed this because github action ( testing ) failed.

Okay, I now see *why* the github action failed.  i.e. because github themselves have removed the py35 options.  We could work around that, but there's probably no point now.  It's fine to remove 3.5 from the github workflow actions.  Please could you explain a bit more *why* you are removing things? e.g. "I removed this because the github action failed because github have removed py35 support from their default workflows."

**Comment by ajkavanagh** [comment]
Okay, so re: verification, it would be good to see a charm that uses this?  e.g. link to charm code that will use this (and probably what you are testing with?)  It's just *really* hard to see how this works without seeing it in use.  Thanks!

**Comment by xtrusia** [comment]
> Okay, so re: verification, it would be good to see a charm that uses this? e.g. link to charm code that will use this (and probably what you are testing with?) It's just _really_ hard to see how this works without seeing it in use. Thanks!

hey, it is new feature so please refer to this
this code should be merged for fixing my case.
I think there is no other charms using this.

https://github.com/juju-solutions/interface-prometheus-manual/pull/6/files

**Comment by ajkavanagh** [comment]
> > Okay, so re: verification, it would be good to see a charm that uses this? e.g. link to charm code that will use this (and probably what you are testing with?) It's just _really_ hard to see how this works without seeing it in use. Thanks!
> 
> hey, it is new feature so please refer to this this code should be merged for fixing my case. I think there is no other charms using this.
> 
> https://github.com/juju-solutions/interface-prometheus-manual/pull/6/files

Okay, let's finish this off; if you strip out the py3.5 github workflow test (as it's no longer supported, as you originally discovered), then we'll merge this.  Thanks.
